core/psh/login.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Phoenix-RTOS
+ *
+ * login - allows/disallows access to Phoenix SHell 
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Mateusz Niewiadomski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "psh.h"
+
+
+int psh_login(void){
+	char c, *usrpasswd;
+	unsigned int passsz = 0;
+
+	/* allocate memory for user-given password */
+	if ((usrpasswd = calloc(strlen(psh_common.passwd),sizeof(char)+1)) == NULL )
```

Comment:
Is using dynamic memory necessary?
I remember some time ago we were talking with @Maxez and @pawelpisarczyk, that we would like to get rid of mallocs in psh, because of memory shortage on NOMMU architectures. I don't know, if it is still relevant.

Reply:
also: invalid allocation length:
it's `strlen(passwd) * (sizeof(char) + 1)`, so almost 2 times too big.

Reply:
You are right, this approach is bad. It is going not to be dynamic allocation in next login feature iteration Thanks.